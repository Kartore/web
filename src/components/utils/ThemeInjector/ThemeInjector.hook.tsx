import { useTernaryDarkMode } from 'usehooks-ts';
import { useEffect } from 'react';

export const useThemeInjector = () => {
  const { isDarkMode } = useTernaryDarkMode();
  useEffect(() => {
    if (isDarkMode) {
      document.documentElement.classList.add('dark');
    } else {
      document.documentElement.classList.remove('dark');
    }
  }, [isDarkMode]);
};
