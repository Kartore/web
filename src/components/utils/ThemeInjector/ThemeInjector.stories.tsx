import { ThemeInjector } from '.';
import { Meta, StoryObj } from '@storybook/react';
import { expect } from '@storybook/test';

const meta = {
  component: ThemeInjector,
} satisfies Meta<typeof ThemeInjector>;

export default meta;

type Story = StoryObj<typeof meta>;

export const NothingRenderCheck: Story = {
  args: {},
  play: async ({ canvasElement }) => {
    await expect(canvasElement.children.length).toBe(0);
  },
};
