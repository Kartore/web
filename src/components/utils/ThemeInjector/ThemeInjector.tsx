import { useThemeInjector } from '~/components/utils/ThemeInjector/ThemeInjector.hook.tsx';

export const ThemeInjector = () => {
  useThemeInjector();

  return null;
};
