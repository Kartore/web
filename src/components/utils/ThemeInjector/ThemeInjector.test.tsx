import { describe } from 'vitest';
import { render, renderHook } from '@testing-library/react';
import { composeStories } from '@storybook/react';

import * as Stories from './ThemeInjector.stories';
import { useThemeInjector } from '~/components/utils/ThemeInjector/ThemeInjector.hook.tsx';

import * as modules from 'usehooks-ts';
import { vi } from 'vitest';

const { NothingRenderCheck } = composeStories(Stories);

describe('Component: ThemeInjector', () => {
  describe('Snapshot', () => {
    it('NothingRenderCheck', () => {
      const { asFragment } = render(<NothingRenderCheck />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe('Play', () => {
    it('NothingRenderCheck', async () => {
      const { container } = render(<NothingRenderCheck />);
      await NothingRenderCheck.play!({ canvasElement: container });
    });
  });
});

vi.mock('usehooks-ts', () => {
  const originalModule = vi.importActual('usehooks-ts');
  return {
    ...originalModule,
    useTernaryDarkMode: vi.fn().mockReturnValue({
      isDarkMode: false,
      setTernaryDarkMode: vi.fn(),
      ternaryDarkMode: 'system',
      toggleTernaryDarkMode: vi.fn(),
    }),
  };
});

describe('Hook: useThemeInjector', () => {
  it('Inject Dark', async () => {
    vi.spyOn(modules, 'useTernaryDarkMode').mockReturnValue({
      isDarkMode: true,
      setTernaryDarkMode: vi.fn(),
      ternaryDarkMode: 'dark',
      toggleTernaryDarkMode: vi.fn(),
    });
    renderHook(() => useThemeInjector());

    expect(document.documentElement.classList.contains('dark')).toBeTruthy();
  });
  it('Inject Light', async () => {
    vi.spyOn(modules, 'useTernaryDarkMode').mockReturnValue({
      isDarkMode: false,
      setTernaryDarkMode: vi.fn(),
      ternaryDarkMode: 'light',
      toggleTernaryDarkMode: vi.fn(),
    });
    renderHook(() => useThemeInjector());

    expect(document.documentElement.classList.contains('dark')).toBeFalsy();
  });
});
