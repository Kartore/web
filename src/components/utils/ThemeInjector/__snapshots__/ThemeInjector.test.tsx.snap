// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: ThemeInjector > Snapshot > NothingRenderCheck 1`] = `<DocumentFragment />`;
